chanbackup/pubsub.go

```diff
@@ -321,6 +375,38 @@ func (s *SubSwapper) backupUpdater() {
 					err)
 			}
 
+		// We received a manual update. Handle it and update the file.
+		case manualUpdate := <-s.manualUpdates:
+			oldStateSize := len(s.backupState)
+
+			// For all open channels, we'll create a new SCB given
+			// the required information.
+			for _, single := range manualUpdate.singles {
+				log.Debugf("Manual update of channel %v",
+					single.FundingOutpoint)
+
+				s.backupState[single.FundingOutpoint] = single
+			}
+
+			newStateSize := len(s.backupState)
+
+			log.Infof("Updating on-disk multi SCB backup: "+
+				"num_old_chans=%v, num_new_chans=%v",
+				oldStateSize, newStateSize)
+
+			// With out new state constructed, we'll, atomically
+			// update the on-disk backup state.
+			err := s.updateBackupFile()
+			if err != nil {
+				log.Errorf("unable to update backup file: %v",
+					err)
+			}
+
+			// Send the outcome back to the caller of ManualUpdate.
+			// The error channel must have capacity of 1 not to
+			// block here.
+			manualUpdate.errChan <- err
```

Comment:
I think we should send the `err` back in the `if err != nil }` above and then just do `close(manualUpdate.errChan)` to signify success.

Reply:
Done.